--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -464,11 +464,15 @@
 		 * TODO: accept the key if we have a station entry and
 		 *       add it to the device after the station.
 		 */
-		if (!sta || !test_sta_flag(sta, WLAN_STA_ASSOC)) {
-			ieee80211_key_free_unused(key);
-			err = -ENOENT;
-			goto out_unlock;
-		}
+                if (!sta) {
+                        sdata_info(sdata, "mwv1 - no sta\n");
+                        ieee80211_key_free_unused(key);
+                        err = -ENOENT;
+                        goto out_unlock;
+                }
+                if (!test_sta_flag(sta, WLAN_STA_ASSOC)) {
+                        sdata_info(sdata, "mwv2 - no sta assoc\n");
+                }
 	}
 
 	switch (sdata->vif.type) {
